File: back/src/main/java/com/back/global/exception/GlobalExceptionHandler.java
package com.back.global.exception;

import com.back.global.rsData.RsData;
import jakarta.servlet.http.HttpServletResponse;
import lombok.RequiredArgsConstructor;
import org.springframework.web.bind.annotation.ExceptionHandler;
import org.springframework.web.bind.annotation.RestControllerAdvice;

@RestControllerAdvice
@RequiredArgsConstructor
public class GlobalExceptionHandler {
    @ExceptionHandler(com.back.global.exception.ServiceException.class)
    public RsData<Void> handle(ServiceException ex, HttpServletResponse response) {
        RsData<Void> rsData = ex.getRsData();

        response.setStatus(rsData.statusCode());

        return rsData;
    }
}

File: back/src/main/kotlin/com/back/global/exception/GlobalExceptionHandler.kt
package com.back.global.exception

import com.back.global.rsData.RsData
import jakarta.servlet.http.HttpServletResponse
import org.springframework.web.bind.MethodArgumentNotValidException
import org.springframework.web.bind.annotation.ExceptionHandler
import org.springframework.web.bind.annotation.RestControllerAdvice

@RestControllerAdvice
class GlobalExceptionHandler {

    @ExceptionHandler(ServiceException::class)
    fun handle(ex: ServiceException, response: HttpServletResponse): RsData<Void> =
        ex.rsData.also { response.status = it.statusCode }

    @ExceptionHandler(MethodArgumentNotValidException::class)
    fun handle(ex: MethodArgumentNotValidException, response: HttpServletResponse): RsData<Void> {
        val message = ex.bindingResult.fieldErrors.firstOrNull()?.defaultMessage ?: "잘못된 입력입니다."
        return RsData<Void>(ErrorCode.BAD_REQUEST.resultCode, message).also {
            response.status = it.statusCode
        }
    }
}
